### ISSUE TITLE
GET requests that bridge between online meetings and underlying events possible?

### ISSUE THREAD

**Issue by AlexSapple** [microsoftgraph/microsoft-graph-docs#15537]
Hi, 

considering the following scenario (please bear with me for long explanation below);

the tldr: is using a joinUrl or onlineMeetingId, can I access data from the /events endpoint (where that endpoint returns more details than the /onlineMeetings endpoint)

let's say a user has created an online meeting in Teams and is using a custom app within Teams - that custom app has access to a context that include Ids that look like this; 
> MCMxOTptZWV0aW5nX056TXlZbVkyTTJZdE1HTmpNQzAwTW1SbUxXSTJaV0V0TlRKaVptTXlZamN4T1dRNEB0aHJlYWQudjIjMA== 

which can be base64 decoded to;

> 0#19:meeting_NzMyYmY2M2YtMGNjMC00MmRmLWI2ZWEtNTJiZmMyYjcxOWQ4@thread.v2#0

the custom app wants to pass this off to custom software and so passes this Id - The custom software wants to query information about this meeting and wants to do something like; 

`https://graph.microsoft.com/v1.0/me/onlineMeetings/0#19:meeting_NzMyYmY2M2YtMGNjMC00MmRmLWI2ZWEtNTJiZmMyYjcxOWQ4@thread.v2#0`

though this seems to return 404's - maybe related to https://github.com/microsoftgraph/microsoft-graph-docs/issues/11696

but ok, so second attempt, the custom Teams app probably has access to the join Url and can pass that instead allowing the custom querying along the lines 

`https://graph.microsoft.com/v1.0/me/onlineMeetings?$filter=joinweburl eq 'https://teams.microsoft.com/l/meetup-join/19%3ameeting_NzMyYmY2M2YtMGNjMC00MmRmLWI2ZWEtNTJiZmMyYjcxOWQ4%40thread.v2/0?context=%7b%22Tid%22%3a%222e95380c-3746-4ce0-a174-d126efcb0187%22%2c%22Oid%22%3a%221d39395c-7a87-49fd-ad9e-3ce3fcdf1744%22%7d'`

which is great - but there are various bits of info that it doesn't return (that I want) - things like ; 

> IsCancelled, Organizer, OriginalTimezones, Locations 

The events endpoint has all this lovely extra information (and when using Graph to create events, there is a little bridge to make the event have a meeting associated - the `IsOnlineMeeting` flag

is there any mechanism to go the other way - so starting from an online meeting to extract Event data? or is this just not how it works?

I've already tried hooking in to it with the joinUrl like this; 
`https://graph.microsoft.com/v1.0/me/events?$filter=onlinemeeting/joinUrl eq 'https://teams.microsoft.com/l/meetup-join/19%3ameeting_NzMyYmY2M2YtMGNjMC00MmRmLWI2ZWEtNTJiZmMyYjcxOWQ4%40thread.v2/0?context=%7b%22Tid%22%3a%222e95380c-3746-4ce0-a174-d126efcb0187%22%2c%22Oid%22%3a%221d39395c-7a87-49fd-ad9e-3ce3fcdf1744%22%7d'`

but sadly that fails with "The property 'OnlineMeeting' does not support filtering."

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 06f89278-6b03-e6e4-66a3-2127a11750fe
* Version Independent ID: f6f92861-6d72-2260-2277-d963793fbce2
* Content: [Get onlineMeeting - Microsoft Graph v1.0](https://docs.microsoft.com/en-us/graph/api/onlinemeeting-get?view=graph-rest-1.0&tabs=http)
* Content Source: [api-reference/v1.0/api/onlinemeeting-get.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/main/api-reference/v1.0/api/onlinemeeting-get.md)
* Product: **cloud-communications**
* Technology: **microsoft-graph**
* GitHub Login: @mkhribech
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by AlexSapple** [comment]
correction: organizer is present in Online meetings (my bad) - so that just leaves the cancelled status, locations and original timezones. I'm guessing that the location resources are listed as participants, so I can probably work around that.

**Comment by AlexSapple** [comment]
so.. with a great deal of fighting, I do have a bridge, since the data available in the Teams context is in fact a base64 encoded chat Id (with an extra 0# at the start and #0 at the end), it's possible to do the following

1. decode the base 64 value, and remove the starting and ending chars
2. call to the chat endpoint passing this Id
3. taking the created date field from the response (this is a nice hack because when an online meeting is created, a chat is also created with a very close created date value
4. make a call to the uses events using a filter on createddatetime to create a range that captures the event using the value above
5. from the event responses, do some horror regex against the joinUrl to extract the chat Id from it (hint, the joinUrl contains the chat Id) and check if that matches the original known chat Id - if it does then it's a match

easy - now we have gone from the data available from the context in the MS Teams custom app (which is in fact a chatId rather than an online meeting Id) and we can force our way to getting back to the detailed information of the event.. 

wow though, would be nice if it were possible to do this without jumping through so many hoops

**Comment by frankpeng7** [comment]
Hi @AlexSapple, if I'm understanding this correctly, you were asking if there's a way to get the event information of an online meeting when it's created via a Teams app, is that right? Do you get other information than the following id when creating the meeting using an app with Teams?
> MCMxOTptZWV0aW5nX056TXlZbVkyTTJZdE1HTmpNQzAwTW1SbUxXSTJaV0V0TlRKaVptTXlZamN4T1dRNEB0aHJlYWQudjIjMA==

**Comment by AlexSapple** [comment]
Hi @frankpeng7 

yes, so what I get from the Teams app turns out to be the ChatId (once you base64 decode it) - so I have a model that works as described above, but it's not very nice (and oddly enough, on some occassions - the Chat has a created date that appears to be wildly different from the event creation date - though I'd have thought they would be created at almost the same time - I have on occasion seen chats that appear to have been created a day or so in the past of the event they relate to. 

I've not actually used the online meeting endpoint as it has less rich data than the event. 

ultimately this can probably be closed as I have a workaround - but the takeaway would be that it would be great if there was an easier way of going from the Id exposed by the Teams app context and the underlying event entity that it relates too.

**Comment by frankpeng7** [comment]
Hi @AlexSapple I'd like to confirm this - when you initially said
> let's say a user has created an online meeting in Teams 

did you create the meeting via /onlineMeetings Graph API?

**Comment by AlexSapple** [comment]
HI @frankpeng7,

Not quite, I refer to when a meeting has been created in the normal fashion in Teams client, but has attached my custom app as a tab. The custom app then wants to handoff some information to my backend about the created meeting and so makes a call out to my endpoint. The tricky part is that the information exposed by the Teams client to the custom app (the Teams context) does not have the level of detail I require (and in fact appears to use a ChatId as identifier for the meeting.  So my backend has to deal with the chatId and find a way to lookup the rest of the information (which is where GraphAPI comes in). 

things I guess I'm unclear about;

ChatId vs OnlineMeetingId vs EventId - these all refer to different entities but are there clear ways to jump between the 3 of them. The meeting invitation is perhaps the way to go, but it's not ideal because if our context is ChatId, then as far as I can see, the only way to get the the onlineMeeting or Event is via some nasty regex parsing of the invitation to match the ChatId which is encoded in the invitation. This is ultimately what I am doing for now but it isn't a nice solution.

Does this make sense?

**Comment by frankpeng7** [comment]
@AlexSapple Gotcha. Unfortunately, I'm working on the /onlineMeetings Graph API which didn't come into this picture. In Graph, an event is not necessarily associated with an online meeting that's why they are two resources and using different IDs. Right now you can query an online meeting using the joinUrl provided in the event, but the way back is yet to support as we rarely got such feature requests. As for the bot SDK you're using, I'm afraid currently there's no easy way to query event with the bot context you are provided with, but I'll forward this request to relevant team!
